_posts/2023-08-04-deploying-OIDC-webapp-on-openshift.adoc

```diff
@@ -0,0 +1,161 @@
+```
+docker build -t quay.io/<your-quay-username>/wildfly:latest .
+docker push quay.io/<your-quay-username>/wildfly:latest
+```
+
+=== Creating a helm release 
+
+Lastly, we will be creating a helm release of our local WildFly image and deploying our application on it using a heml chart. Navigate to `charts` directory and open the `helm.yaml` file. The contents of the file will be as follows: 
+```
+image: 
+  name: quay.io/<Username>/wildfly
+build:
+  enabled: false
+deploy:
+  env:
+    - name: OIDC_PROVIDER_URL
+      value: 'https://sso-some-user.apps.sandbox-m4.g2pi.p1.openshiftapps.com'
+    - name: WILDFLY_OVERRIDING_ENV_VARS
+      value: "1"
+    - name: SUBSYSTEM_LOGGING_PATTERN_FORMATTER_COLOR_PATTERN__PATTERN
+      value: "[redhat-openshift] %-5p %s%e%n"
+```
+There are three fields that we have specified: image, build and deploy. 
+
+Using the image tag, we are specifying that we want the pod to use the image of our application that we just built. Notice that we did not specify the tag. This is because when the helm chart uploads, the `latest` tag is appended to it by default, which is why we tagged our application image as that. If you want to use a different tag, then you can edit this by clicking on the pod, click on `Edit Deployment` and changing it under `Image Name`. 
+
+We have also added some environment variable. Notice the `OIDC_PROVIDER_URL`. The value of this will be the url for your RH-SSO pod. So, copy that value and paste it as the value. 
+
+Change the image name on the `charts/helm.yaml` file to `quay.io/<your-quay-username>/wildfly` and use the following commands to create a cloud instance for your webapp using the wildfly helm chart: 
+```
+helm install simple-webapp-oidc -f charts/helm.yaml wildfly/wildfly
+```
+Click on the `simple-webapp-oidc` pod and under the `resources` tag, you should see the status of your deployment. 
+
+=== Importing Kubernetes Secret
+
+If you see a image pull error, then navigate to your quay console, click on `Account Settings`, `Generate Encrypted Password` and log in. Under the `Kubernetes Secret` tab you will be able to download the yaml file for your Kubernetes secret. Upload it to your OpenShift Secrets by going to the `Secret` tab on Developer Sandbox, clicking on `Create`, `From Yaml` and replacing it with the contents of your Kubernetes Secret. Now you should be able to see your pod run and look at the logs my clicking on `View logs`.  
+
+Alternatively you can link your pull secret so OpenShift can automatically use your pull secret to obtain the image when needed. You can learn more about handling secrets on OpenShift on https://docs.openshift.com/container-platform/4.10/openshift_images/managing_images/using-image-pull-secrets.html[their documentation]. 
```

Comment:
Ah, might be good to mention this at the beginning of this section.

Reply:
I think maybe the header for this is not clear. 
Perhaps I can change the heading from `Importing Kubernetes Secret` to `Resolving image pull error` to make it clear?

Reply:
I think this section should go before the helm release section if it's a pre-requisite for successfully being able to pull the image from quay.io. I think that should be explained and maybe the title could be something like "Configuring Image Pull Secret" or something like that.

We can mention that if they've already done this they don't need to do it again.

Reply:
Fixed!